$OpenBSD: patch-lib_sonos_discovery_rb,v 1.2 2014/01/22 20:39:04 jeremy Exp $

IP_MULTICAST_TTL socket option on OpenBSD currently doesn't accept
int, only unsigned char.

--- lib/sonos/discovery.rb.orig	Wed Dec 31 16:00:00 1969
+++ lib/sonos/discovery.rb	Wed Jan 22 10:20:28 2014
@@ -75,7 +75,7 @@ module Sonos
       @socket = UDPSocket.open
 
       # We're going to use IP with the multicast TTL. Mystery third parameter is a mystery.
-      @socket.setsockopt(Socket::IPPROTO_IP, Socket::IP_MULTICAST_TTL, 2)
+      @socket.setsockopt(Socket::Option.new(:INET, :IPPROTO_IP, :IP_MULTICAST_TTL, 2.chr))
     end
 
     def search_message
